dandiapi/api/services/asset/__init__.py

```diff
@@ -135,20 +135,20 @@ def add_asset_to_version(
     if zarr_archive and zarr_archive.dandiset != version.dandiset:
         raise ZarrArchiveBelongsToDifferentDandisetError
 
-    if isinstance(asset_blob, EmbargoedAssetBlob):
-        embargoed_asset_blob = asset_blob
-        asset_blob = None
-    else:
-        embargoed_asset_blob = None
-        asset_blob = asset_blob  # noqa: PLW0127
-
+    # Creating an asset in an OPEN dandiset that points to an embargoed blob results in that
+    # blob being un-embargoed
+    unembargo_asset_blob = (
+        asset_blob is not None
+        and asset_blob.embargoed
+        and version.dandiset.embargo_status == Dandiset.EmbargoStatus.OPEN
+    )
```

Comment:
I'm trying to think if there's any negative implications in terms of security here. 

Say an `Asset` is part of an embargoed dandiset. Then, someone without permissions to view that dandiset comes along and creates a new asset on their own open dandiset and associates it with the embargoed asset's `AssetBlob`. That will cause the blob to be unembargoed without the embargoed dandiset owner's consent, right?

Reply:
In the situation you've described, the second person that comes along and creates a new asset in an open dandiset doesn't have permissions to view the first embargoed dandiset. Since they don't have permissions on that dandiset, they wouldn't have permissions to view any of the assets of that dandiset either, and so wouldn't know the `blob_id` of the embargoed `AssetBlob`. Thus, they wouldn't be able to construct a new asset using that embargoed blob's `blob_id`.

The only other way someone would be able to get that embargoed assetblob's `blob_id` is if they were to attempt to upload the same file again, resulting in the de-duplication system pointing them to the existing assetblob, instead of allowing the upload to continue. That's still okay, since in that case, both people have access to the file in question, and could each do whatever they wanted to with it.

Reply:
Makes sense, thanks for clarifying!